rtos/Thread.h

```diff
@@ -317,13 +317,28 @@ class Thread : private mbed::NonCopyable<Thread> {
     */
     static osEvent signal_wait(int32_t signals, uint32_t millisec=osWaitForever);
 
-    /** Wait for a specified time period in millisec:
+    /** Wait for a specified time period in milliseconds
+      Being tick-based, the delay will be up to the specified time - eg for
+      a value of 1 the system waits until the next millisecond tick occurs,
+      leading to a delay of 0-1 milliseconds.
       @param   millisec  time delay value
       @return  status code that indicates the execution status of the function.
       @note not callable from interrupt
+      @note this call is equivalent to Kernel::delay()
     */
     static osStatus wait(uint32_t millisec);
 
+    /** Wait until a specified time in millisec
+      The specified time is according to Kernel::get_ms_count().
+      @param   millisec absolute time in millisec
+      @return  status code that indicates the execution status of the function.
+      @note not callable from interrupt
+      @note if the time in the past, or >= 2^32-1 ms in the future, this
+            returns with an immediate error
+      @note this call is equivalent to Kernel::delay_until()
+    */
+    static osStatus wait_until(uint64_t millisec);
```

Comment:
What are you thoughts on dropping the Kernel class and adding get_ms_count() to `Thread` since the other timing functions are already here? The Kernel class could always be added later if kernel specific functionality is needed. This would also keep the API expansion to a minimum.

Reply:
That was what my first draft looked like, but changed to this after @0xc0170's comments, which I tended to agree with.

I'm struggling a bit to find logic in CMSIS-RTOS's placement of stuff in osThread versus osKernel versus osDelay - why not "osThreadLock" and "osKernelDelay" and "osYield"?

Although, thinking harder I guess delay is specifically current thread only, so actually delay probably /shouldn't/ be in Kernel - delaying one thread doesn't affect anything else in the system, unlike the lock. I'm now inclined to take those out of Kernel.